code/xcvm/cosmwasm/contracts/order/src/lib.rs

```diff
@@ -263,17 +266,70 @@ impl OrderContract<'_> {
 	}
 
 	#[msg(exec)]
-	pub fn route(&self, ctx: ExecCtx, _msg: RouteSubMsg) -> StdResult<Response> {
+	pub fn route(&self, ctx: ExecCtx, msg: RouteSubMsg) -> StdResult<Response> {
 		ensure!(
 			ctx.info.sender == ctx.env.contract.address,
 			StdError::GenericErr { msg: "only self can call this".to_string() }
 		);
+
 		ctx.deps.api.debug(
 			"so here we add route execution tracking to storage and map route to CVM program",
 		);
+
+		let _cvm = Self::traverse_route(msg.route);
+
 		Ok(Response::default())
 	}
 
+	/// converts high level route to CVM program
+	fn traverse_route(route: ExchangeRoute) -> cvm::shared::XcProgram {
+		let mut program = XcProgram {
+			tag: b"may be use solution id and some chain for tracking".to_vec(),
+			instructions: vec![],
+		};
+
+		let mut exchanges = Self::traverse_exchanges(route.exchanges);
+		program.instructions.append(&mut exchanges);
+
+		let mut spawns = Self::traverse_spawns(route.spawns);
+		program.instructions.append(&mut spawns);
+
+		program
+	}
+
+	fn traverse_spawns(spawns: Vec<Spawn<ExchangeRoute>>) -> Vec<cvm::shared::XcInstruction> {
+		let mut result = vec![];
+		for spawn in spawns {
+			let spawn = if let Some(execute) = spawn.execute {
+				let program = Self::traverse_route(execute);
+				XcInstruction::Spawn {
+					network_id: spawn.to_chain.into(),
+					salt: b"solution".to_vec(),
+					assets: <_>::default(), // map spawn.carry to CVM assets
+					program,
+				}
+			} else {
+				XcInstruction::Spawn {
+					network_id: spawn.to_chain.into(),
+					salt: b"solution".to_vec(),
+					assets: <_>::default(), // map spawn.carry to CVM assets
+					program: XcProgram {
+						tag: b"solution".to_vec(),
+						instructions: vec![], // we really just do final transfer
+					},
+				}
+			};
+			result.push(spawn);
+		}
+		result
```

Comment:
The Spawn instruction differ in only program so the `if` statement can
be shorter with less duplication.  Also, this could be collect:

```suggestion
		spawns.into_iter().map(|spawn| {
			let program = if let Some(execute) = spawn.execute {
				Self::traverse_route(execute)
			} else {
				XcProgram {
					tag: b"solution".to_vec(),
					instructions: vec![], // we really just do final transfer
				}
			};
			XcInstruction::Spawn {
				network_id: spawn.to_chain.into(),
				salt: b"solution".to_vec(),
				assets: <_>::default(), // map spawn.carry to CVM assets
				program,
			}
		}).collect()
```